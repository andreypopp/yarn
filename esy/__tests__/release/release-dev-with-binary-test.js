/**
 * @flow
 */

jest.setTimeout(200000);

import * as path from 'path';
import {
  createProject,
  readDirectory,
  cleanUp,
  esyBin,
  runIn,
  run,
  file,
  directory,
  packAndNpmInstallGlobal,
  initFixture,
} from './utils';

afterEach(cleanUp);

test('esy release dev: with "releasedBinaries"', async function() {
  const fixture = await initFixture(path.join(__dirname, 'fixtures', 'with-binary'));
  await runIn(fixture.project, 'npm', 'install');
  await runIn(fixture.project, esyBin, 'release', 'dev');

  expect(await readDirectory(fixture.project, '_release')).toMatchSnapshot('release');

  await packAndNpmInstallGlobal(fixture, '_release', 'dev');

  expect(await readDirectory(fixture.npmPrefix)).toMatchSnapshot('installation');

  const res = await run(path.join(fixture.npmPrefix, 'bin', 'say-hello.exe'));
  expect(res).toBe('HELLO');
});
