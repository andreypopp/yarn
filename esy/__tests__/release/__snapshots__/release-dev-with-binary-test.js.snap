// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`installation 1`] = `
"<root>
| bin
| | say-hello.exe
| | with-binary
| etc
| lib
| | node_modules
| | | with-binary
| | | | .bin
| | | | | say-hello.exe
| | | | | with-binary
| | | | package.json
| | | | postinstall.sh
| | | | prerelease.sh
| | | | records
| | | | | recordedClientBuildStorePath.txt
| | | | | recordedClientInstallStorePath.txt
| | | | | recordedCoppiedArtifacts.txt
| | | | | recordedServerBuildStorePath.txt
| | | | rel
| | | | | 3.x.x__________________________
| | | | | | i
| | | | | | | with_binary-0.1.0
| | | | | | | | bin
| | | | | | | | | say-hello.exe
| | | | | | | | doc
| | | | | | | | etc
| | | | | | | | lib
| | | | | | | | man
| | | | | | | | sbin
| | | | | | | | share
| | | | | node_modules
| | | | | | .cache
| | | | | | | _esy
| | | | | | | | build-eject
| | | | | | | | | bin
| | | | | | | | | | fastreplacestring.cpp
| | | | | | | | | | fastreplacestring.exe
| | | | | | | | | | realpath
| | | | | | | | | | realpath.c
| | | | | | | | | | render-env
| | | | | | | | | | runtime.sh
| | | | | | | | | command-env
| | | | | | | | | command-env.in
| | | | | | | | | eject-env
| | | | | | | | | final-install-path-set.txt
| | | | | | | | | final-install-path-set.txt.in
| | | | | | | | | Makefile
| | | | | | | | | sandbox.sb.in
| | | | | | | | store
| | | | | | | | | b
| | | | | | | | | i
| | | | | | | | | s
| | | | | package.json
| | | | | say-hello.exe
| | | | | yarn.lock"
`;

exports[`release 1`] = `
"<root>
| dev
| | .bin
| | | say-hello.exe
| | | with-binary
| | package.json
| | postinstall.sh
| | prerelease.sh
| | rel
| | | package.json
| | | say-hello.exe"
`;
